src/platform/Zephyr/WiFiManager.cpp

```diff
@@ -53,11 +75,38 @@ CHIP_ERROR WiFiManager::Init()
         ChipLogError(DeviceLayer, "wpa_supplicant is not initialized!");
         return CHIP_ERROR_INTERNAL;
     }
-    else
-    {
-        ChipLogDetail(DeviceLayer, "wpa_supplicant has been initialized");
+
+    // TODO: consider moving these to ConnectivityManagerImpl to be prepared for handling multiple interfaces on a single device.
+    Inet::UDPEndPointImplSockets::SetJoinMulticastGroupHandler([](Inet::InterfaceId interfaceId, const Inet::IPAddress & address) {
+        const in6_addr addr = ToZephyrAddr(address);
+        net_if * iface      = GetInterface(interfaceId);
+        VerifyOrReturnError(iface != nullptr, INET_ERROR_UNKNOWN_INTERFACE);
+
+        net_if_mcast_addr * maddr = net_if_ipv6_maddr_add(GetInterface(interfaceId), &addr);
+
+        if (maddr && !net_if_ipv6_maddr_is_joined(maddr) && !net_ipv6_is_addr_mcast_link_all_nodes(&addr))
+        {
+            net_if_ipv6_maddr_join(maddr);
+        }
+
         return CHIP_NO_ERROR;
-    }
+    });
+
+    Inet::UDPEndPointImplSockets::SetLeaveMulticastGroupHandler([](Inet::InterfaceId interfaceId, const Inet::IPAddress & address) {
+        const in6_addr addr = ToZephyrAddr(address);
+        net_if * iface      = GetInterface(interfaceId);
+        VerifyOrReturnError(iface != nullptr, INET_ERROR_UNKNOWN_INTERFACE);
+
+        if (!net_ipv6_is_addr_mcast_link_all_nodes(&addr) && !net_if_ipv6_maddr_rm(net_if_get_default(), &addr))
```

Comment:
Again, why not using the `iface` instead of `net_if_get_default()`?

Reply:
Right, thanks!